packages/terminal/src/browser/terminal-preferences.ts

```diff
@@ -126,7 +153,13 @@ export interface TerminalConfiguration {
     'terminal.integrated.copyOnSelection': boolean,
     'terminal.integrated.cursorBlinking': boolean,
     'terminal.integrated.cursorStyle': CursorStyleVSCode,
-    'terminal.integrated.cursorWidth': number
+    'terminal.integrated.cursorWidth': number,
+    'terminal.integrated.shell.windows': string,
+    'terminal.integrated.shell.osx': string,
+    'terminal.integrated.shell.linux': string,
+    'terminal.integrated.shellArgs.windows': string,
+    'terminal.integrated.shellArgs.osx': string,
+    'terminal.integrated.shellArgs.linux': string
```

Comment:
The following preferences do not match their definitions. 
They support an array of elements, and here we define them as a `string` only.

Reply:
Fixed